src/Components/Components/src/Forms/InputComponents/InputNumber.cs

```diff
@@ -0,0 +1,173 @@
+            }
+            else if (targetType == typeof(double))
+            {
+                _parser = TryParseDouble;
+            }
+            else if (targetType == typeof(decimal))
+            {
+                _parser = TryParseDecimal;
+            }
+            else
+            {
+                throw new InvalidOperationException($"The type '{targetType}' is not a supported numeric type.");
+            }
+        }
+
+        [Parameter] string ParsingErrorMessage { get; set; } = "The {0} field must be a number.";
+
+        /// <inheritdoc />
+        protected override void BuildRenderTree(RenderTreeBuilder builder)
+        {
+            base.BuildRenderTree(builder);
+            builder.OpenElement(0, "input");
+            builder.AddAttribute(1, "class", FieldClass);
+            builder.AddAttribute(2, "value", BindMethods.GetValue(CurrentValueAsString));
+            builder.AddAttribute(3, "onchange", BindMethods.SetValueHandler(__value => CurrentValueAsString = __value, CurrentValueAsString));
+            builder.CloseElement();
+        }
+
+        /// <inheritdoc />
+        protected override bool TryParseValueFromString(string value, out T result, out string validationErrorMessage)
+        {
+            if (_parser(value, out result))
+            {
+                validationErrorMessage = null;
+                return true;
+            }
+            else
+            {
+                validationErrorMessage = string.Format(ParsingErrorMessage, FieldIdentifier.FieldName);
+                return false;
+            }
+        }
+
+        static bool TryParseShort(string value, out T result)
+        {
+            var success = short.TryParse(value, out var parsedValue);
+            if (success)
+            {
+                result = (T)(object)parsedValue;
+                return true;
+            }
+            else
+            {
+                result = default;
+                return false;
+            }
+        }
+
+        static bool TryParseInt(string value, out T result)
+        {
+            var success = int.TryParse(value, out var parsedValue);
+            if (success)
+            {
+                result = (T)(object)parsedValue;
+                return true;
+            }
+            else
+            {
+                result = default;
+                return false;
+            }
+        }
+
+        static bool TryParseLong(string value, out T result)
+        {
+            var success = long.TryParse(value, out var parsedValue);
+            if (success)
+            {
+                result = (T)(object)parsedValue;
+                return true;
+            }
+            else
+            {
+                result = default;
+                return false;
+            }
+        }
+
+        static bool TryParseFloat(string value, out T result)
+        {
+            var success = float.TryParse(value, out var parsedValue);
```

Comment:
Probably you should parse float/double/decimal values with InvariantCulture. I have Polish Windows and Polish culture set in browsers. In browser I see comma as decimal separator but field values in Blazor 0.7 have dot. Example code:
<input type="number" bind="@number" /> @number
@functions
{ string number = "1.9"; }
Notice that input type is set to "number" and contrary to InputDate you don't add this attribute in InputNumber.

Reply:
Good point. I've changed it to `InvariantCulture`. We'll review further for better localizability when we do #5471.